runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7275,6 +7275,11 @@ class INTERPRETER_CLASS
 				rc = THROW_AIOB;
 			} else {
 				j9object_t value = *(j9object_t*)_sp;
+				J9ArrayClass* arrayClass = (J9ArrayClass*)J9OBJECT_CLAZZ(_currentThread, arrayref);
```

Comment:
Also please add these changes under the J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES flag.